File: backend/controllers/authController.js
const bcrypt = require("bcrypt");
const db = require("../db");

exports.register = async (req, res) => {
  const { username, password, role } = req.body;
  try {
    const hashedPassword = await bcrypt.hash(password, 10);
    await db.query(
      "INSERT INTO users (username, password, role) VALUES (?, ?, ?)",
      [username, hashedPassword, role]
    );
    res.status(201).json({ message: "Utilisateur créé !" });
  } catch (err) {
    res.status(500).json({ error: err.message });
  }
};

exports.login = async (req, res) => {
  const { username, password } = req.body;
  try {
    const [users] = await db.query("SELECT * FROM users WHERE username = ?", [
      username,
    ]);
    if (users.length === 0) {
      return res.status(400).json({ error: "Utilisateur non trouvé" });
    }

    const user = users[0];
    const valid = await bcrypt.compare(password, user.password);
    if (!valid) {
      return res.status(400).json({ error: "Mot de passe incorrect" });
    }

    // Pour l'instant simple : on retourne l'ID (pas de token encore)
    res.json({
      message: "Connexion réussie",
      userId: user.id,
      role: user.role, // ✅ très important pour rediriger correctement
    });
    
  } catch (err) {
    res.status(500).json({ error: err.message });
  }
};

File: backend/controllers/auth_controller.py
from flask import Blueprint, request, jsonify
import bcrypt
from db import get_connection
import mysql.connector

auth_bp = Blueprint('auth', __name__, url_prefix='/auth')

@auth_bp.route('/register', methods=['POST'])
def register():
    data = request.json
    username = data.get('username')
    password = data.get('password')
    role = data.get('role')

    if not all([username, password, role]):
        return jsonify(error="Champs manquants"), 400

    hashed = bcrypt.hashpw(password.encode('utf-8'), bcrypt.gensalt())
    conn = get_connection()
    cursor = conn.cursor()
    try:
        cursor.execute(
            "INSERT INTO users (username, password, role) VALUES (%s, %s, %s)",
            (username, hashed.decode('utf-8'), role)
        )
        conn.commit()
        return jsonify(message="Utilisateur créé !"), 201
    except mysql.connector.Error as e:
        return jsonify(error=str(e)), 500
    finally:
        cursor.close()
        conn.close()

@auth_bp.route('/login', methods=['POST'])
def login():
    data = request.json
    username = data.get('username')
    password = data.get('password')

    conn = get_connection()
    cursor = conn.cursor(dictionary=True)
    try:
        cursor.execute("SELECT * FROM users WHERE username = %s", (username,))
        user = cursor.fetchone()
        if not user:
            return jsonify(error="Utilisateur non trouvé"), 400

        if not bcrypt.checkpw(password.encode('utf-8'), user['password'].encode('utf-8')):
            return jsonify(error="Mot de passe incorrect"), 400

        return jsonify(message="Connexion réussie", userId=user['id'])
    except mysql.connector.Error as e:
        return jsonify(error=str(e)), 500
    finally:
        cursor.close()
        conn.close()